### PR TITLE
pkg/libcoap: delete unused function check_opt_size

### PR DESCRIPTION
**Pull request by cgundogan**
This PR adds a patch to delete the unused function and updates all other patches to have a correct commit message numbering

### CODE REVIEW

**Comment by cgundogan** [comment]
fixes #3863

File: pkg/libcoap/0009-delete-unused-function-check_opt_size.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
+From 4e065c420056a5416d9d4b00439042d7adcefddb Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Cenk=20G=C3=BCndo=C4=9Fan?= <cnkgndgn@gmail.com>
```
Comment (cgundogan) [inline pkg/libcoap/0009-delete-unused-function-check_opt_size.patch]: funny..

**Comment by OlegHahm** [comment]
Looks sensible, let's ask Travis.

**Comment by cgundogan** [comment]
fixed white space errors - amended immediately

**Comment by Lotterleben** [comment]
:+1:

**Comment by authmillenon** [comment]
Was that an ACK?

**Comment by Lotterleben** [comment]
If Travis is happy?

**Comment by Lotterleben** [comment]
Strider! Sorry.. /o\

**Comment by authmillenon** [comment]
Strider needs not to be happy, strider needs to be working first ;-P

**Comment by OlegHahm** [comment]
Well, as long as Strider is doing something we could also wait for it.

**Comment by authmillenon** [comment]
Seems like it doesn't do anything...

**Pull request merged**